sshd-common/src/main/java/org/apache/sshd/common/signature/SignatureRSA.java

```diff
@@ -52,8 +52,21 @@
 
     private int verifierSignatureSize = -1;
 
+    private final String sshAlgorithmName;
+
     protected SignatureRSA(String algorithm) {
+        this(algorithm, null);
```

Comment:
Since you are using `ValidateUtils.checkNotNullAndNotEmpty` this constructor is invalid since whoever uses it will get an exception.